t/test_multivalue_separator.rb

```diff
@@ -57,4 +61,10 @@
     doe[:images][0][:url].must_equal 'http://example.org/a.jpg'
     doe[:images][1][:url].must_equal 'http://example.org/b.png'
   end
+
+  it 'should not split URL into multiple entries when separator is legitimate character in URL' do
+    member = australia.data[:persons].find { |p| p[:name] == 'Joanna Gash' }
+    links = member[:links].select { |l| l[:note] == 'website' }
+    links.count.must_equal 1
```

Comment:
I think you need to test that the URL is actually the correct thing. Just counting how many links you get back would pass in any number of broken scenarios.